lib/ecto/schema.ex

```diff
@@ -319,6 +320,16 @@ defmodule Ecto.Schema do
   end
 
   @doc """
+  TODO
+  """
+  defmacro schema(scope, source, [do: block]) do
```

Comment:
I missed this discussion, but yes I'm not a big fan of the module attribute - to be honest I'm not a fan of solving this with the additional scope either - I would love to see some more generic solution in the compiler to prevent situations like this.